app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/rootnav/RootNavViewModel.kt

```diff
@@ -66,6 +67,20 @@ class RootNavViewModel @Inject constructor(
 
             userState?.activeAccount?.needsPasswordReset == true -> RootNavState.ResetPassword
 
+            specialCircumstance is SpecialCircumstance.CompleteRegistration -> {
+                // When the user is on the lock screen or already in the vault
+                if (userState?.activeAccount != null && !authRepository.hasPendingAccountAddition) {
+                    authRepository.hasPendingAccountAddition = true
+                    return
```

Comment:
We can't just return from here, as that will leave the user in a bad state. Why aren't we continuing to the `CompleteOngoingRegistration` state?

Reply:
After the set of `authRepository.hasPendingAccountAddition` to true it will trigger this code again and continue to the `CompleteOngoingRegistration`.

Reply:
💭 I was under the impression this was to prevent an edge case navigating to the `CompleteRegistration` screen if the there was some active session in progress between the time when the registration started and this intent is handled. I think technically I could start this process and then go switch to another account before completing this. Would it make more sense to mutate the `hasPendingAccountAddition` in the `MainViewModel` before pushing the data through the circumstance manager?

Reply:
This is when the user is active and we click the **verify email** in the email that was sent. Setting the `hasPendingAccountAddition` simulates the click of "add new account" from the account switcher. After that it will launch the `Landing` -> `CompleteRegistration`. Having the `hasPendingAccountAddition` also allows the user to go back to `Landing` and use the account switcher and pick another account.

Reply:
If the state is the same, the re-trigger will be ignored, we should just return the correct state everytime:

```kotlin
if (userState?.activeAccount != null) {
    authRepository.hasPendingAccountAddition = true
}
CompleteOngoingRegistration(...)
```

Reply:
Still have an issue here, we cannot return here without addressing the state.

Reply:
Moving the set of `hasPendingAccountAddition` to the `MainViewModel` before firing `CompleteRegistration` special circumstance the did the trick.